### ISSUE TITLE
Attributes with DevInt data_type not supported

### ISSUE THREAD

**Issue by kmadisa** [tango-controls/pytango#138]
I was trying to create an attribute during runtime using the _Attr_ class instance:
`attr = Attr(attribute_name, attr_dtype, rw_type)`
It worked for other data types, however it crashed when the _attr_dtype_ is set to  _DevInt_

```
Exiting: Server exited with PyTango.DevFailed:
DevFailed[
DevError[
    desc = Attribute : Value2:  Data type is not supported
  origin = Attr::check_type
  reason = API_AttrWrongDefined
severity = ERR]
]

Exited
Segmentation fault (core dumped)
```

**Comment by LJBD** [comment]
Hello!

AFAIK, there's no `DevInt` type, there are `DevShort`, `DevLong` and `DevLong64` types (according to [this page](tango-controls.readthedocs.io/en/latest/development/device-api/ds-guideline/TangoDesignGuidelines-Revision-7.html#static-attribute-properties) of Tango Controls documentation).

To specify an attribute's type in PyTango we use Python types. [Here](http://pytango.readthedocs.io/en/latest/data_types.html) is a documentation page regarding which Python type you should use for which of Tango types.

Another thing: it's encouraged to use the so-called "high level" API to create interface of your Tango Device Classes. [Here](http://pytango.readthedocs.io/en/latest/quicktour.html#server) is a quick guide on how to do that.

In short, you should do the following:
```python
class MyDevice(Device):
    _my_attribute = 0

    my_attribute = attribute(dtype=int)

    def read_my_attribute():
        return self._my_attribute
```

**Comment by AntoineDupre** [comment]
Hello,

FYI:

In your example the attribute is not created during runtime.

In PyTango, it is possible to create a dynamic attribute from ```Attr``` object or ```AttrData``` object. In those two cases, there is no python type to tango type conversion, it requires to use directly the tango type:
```python
my_attr = Attr("my_attribute", tango.DevShort, tango.READ_WRITE)
self.add_attribute(
    attr=my_attr, 
    r_meth=a_read_method, 
    w_meth=a_write_method, 
    is_allo_meth=an_allowed_method)
```

Note that it is not possible to use the ```return``` statement in ```r_meth```, ```w_meth``` and ```is_allo_meth``` methods.

To get python to tango type conversion, it might be possible to use the ```tango.server.attribute``` class to create a dynamic attribute (this class inheritates from ```AttrData```).

```python
my_attribute = attribute(name="my_attribute", dtype=int)
self.add_attribute(
    attr=my_attribute,
    r_meth=a_read_method, 
    w_meth=a_write_method, 
    is_allo_meth=an_allowed_method)
```
